Gems/Atom/RPI/Code/Include/Atom/RPI.Public/ViewGroup.h

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#pragma once
+
+#include <Atom/RPI.Public/View.h>
+#include <Atom/RPI.Public/ViewProviderBus.h>
+#include <Atom/RPI.Public/XR/XRRenderingInterface.h>
+#include <AzCore/Name/Name.h>
+
+namespace AZ
+{
+    namespace RPI
+    {
+        //! This class is responsible for managing stereoscopic and non-stereoscopic views.
+        //! It also allows calling code to provide handlers or functions to be called in case the projection
+        //! or view matrix changes. 
+        class ViewGroup final
+        {
+        public:
+            AZ_TYPE_INFO(View, "{C1AD45ED-03E2-41E0-9DC7-779B4B1842FF}");
+            AZ_CLASS_ALLOCATOR(ViewGroup, AZ::SystemAllocator, 0);
+
+            struct Descriptor
+            {
+                //! Function to be called in case view matrix changed
+                AZStd::function<void(AZ::RPI::ViewPtr view)> m_viewEventFunction;
+
+                //! Function to be called in case projection matrix changed
+                AZStd::function<void(AZ::RPI::ViewPtr view)> m_projectionEventFunction;
+            };
+
+            struct ViewData
+            {
+                //! Pointer to the view
+                AZ::RPI::ViewPtr m_view = nullptr;
+
+                //If a View (i.e m_view) related transforms are updated these handlers are updated
+                MatrixChangedEvent::Handler m_onViewMatrixChangedHandler;
+                MatrixChangedEvent::Handler m_onProjectionMatrixChangedHandler;
+
+                //The event handlers above will also signal the following event in case
+                //any handlers from outside are connected to these events
+                MatrixChangedEvent m_projectionMatrixChangedEvent;
+                MatrixChangedEvent m_viewMatrixChangedEvent;
+            };
+
+            ViewGroup();
+            ~ViewGroup();
+
+            //! Initialization which involves caching the descriptor and XR system related properties if applicable.
+            void Init(Descriptor desc);
```

Comment:
Worth passing `const Descriptor&` instead in case the descriptor grows to avoid making unnecessary copies.